tests/unit/transformers/test_base.py

```diff
@@ -52,23 +489,546 @@ def test__set_columns_data_preserves_order_dataframe(self):
             - Data should have the values from columns_data in the same order
             as they were in columns_data.
         """
+        # Setup
         data = pd.DataFrame({
             'a': [1, 2, 3],
             'b': [4, 5, 6],
-            'c': [1, 2, 3]
         }, index=[2, 0, 1])
         columns = ['c', 'd']
         columns_data = pd.DataFrame({
             'c': [7, 8, 9],
             'd': [10, 11, 12]
         })
 
+        # Run
         BaseTransformer._set_columns_data(data, columns_data, columns)
 
+        # Assert
         expected = pd.DataFrame({
             'a': [1, 2, 3],
             'b': [4, 5, 6],
             'c': [7, 8, 9],
             'd': [10, 11, 12]
         }, index=[2, 0, 1])
         pd.testing.assert_frame_equal(data, expected)
+
+    def test__set_columns_1d_array(self):
+        """Test the ``_set_columns_data`` method.
+
+        The method should not reorder the rows from the ``columns_data``
+        parameter if it is a 1d array and the ``data`` has a different index.
+
+        Input:
+            - data will be a DataFrame with a non-sequential index.
+            - columns_data will be a 1d array.
+            - columns will have the column name of the array.
+
+        Expected behavior:
+            - Data should have the values from columns_data in the same order
+            as they were in columns_data.
+        """
+        # Setup
+        data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4, 5, 6],
+        }, index=[2, 0, 1])
+        columns = ['c']
+        columns_data = np.array([7, 8, 9])
+
+        # Run
+        BaseTransformer._set_columns_data(data, columns_data, columns)
+
+        # Assert
+        expected = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4, 5, 6],
+            'c': [7, 8, 9]
+        }, index=[2, 0, 1])
+        pd.testing.assert_frame_equal(data, expected, check_dtype=False)
```

Comment:
@csala I added this check because on Windows data becomes `int32`, while the expected is `int64`.